pkg/apis/hive/v1/gcp/machinepools.go

```diff
@@ -9,6 +9,20 @@ type MachinePool struct {
 	// eg. n1-standard-4
 	InstanceType string `json:"type"`
 
+	// ControlPlaneSubnet defines the subnet where the control plane will be deployed.
+	// The value should be the name of the subnet.
+	// If ommitted one will be created by the installer.
+	ControlPlaneSubnet string `json:"control_plane_subnet,omitempty"`
+
+	// ComputeSubnet defines the subnet where the compute nodes will be deployed.
+	// The value should be the name of the subnet.
+	// If ommitted one will be created by the installer.
+	ComputeSubnet string `json:"compute_subnet,omitempty"`
+
+	// Network specifies an existing VPC where the cluster should be created
+	// rather than provisioning a new one.
+	Network string `json:"network,omitempty"`
```

Comment:
we must enforce that user set `network` and `subnet` both together or neither.

Reply:
Ack :+1: